bitsail-connectors/connector-local-csv/src/main/java/com/bytedance/bitsail/connector/local/csv/option/LocalCsvReaderOptions.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2022 Bytedance Ltd. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.bytedance.bitsail.connector.local.csv.option;
+
+import com.bytedance.bitsail.common.option.ConfigOption;
+import com.bytedance.bitsail.component.format.csv.option.CsvReaderOptions;
+
+import static com.bytedance.bitsail.common.option.ConfigOptions.key;
+import static com.bytedance.bitsail.common.option.ReaderOptions.READER_PREFIX;
+
+public interface LocalCsvReaderOptions extends CsvReaderOptions {

```

Comment:
we should not define a format in any connector, because the connector should be able to read many format.

Reply:
I change to extends `ReaderOptions.BaseReaderOptions` , am I doing this right? I am still confued, could you please explain more? And how to automatically extends exact interface like `CsvReaderOptions` or `JsonReaderOptions`?

Reply:
Hi Jake, let's break it into two parts. `connector` level and `format` level. `connector` responsible for read or write data from certain engine, e.g. `connector-kafka` `connector-redis`. And `format` is the data format we read from this engine. For example, we read data from kafka through `connector-kafka`, and the format of the data is `Json`.
For local file system, the data was store at local file system, so we need `connector-filesystem`, and we can read many different format types, such as CSV file, Json file, Text file e.t.c. 
Regarding to `ReaderOptions`, this is related to `connector`, so it should be `FileSystemReaderOptions`.

Reply:
Hey Gary, your detailed explanation teaches me a lot, thanks!
For issue #112 we may just let `FileSystemReaderOptions` extend `CsvReaderOptions`. In the future we might support other file formats, how do we let `FileSystemReaderOptions` compatible with them? By extending other exclusive interface like below?
```java
public interface FileSystemReaderOptions extends CsvReaderOptions, JsonReaderOptions, TxtReaderOptions {
  ...
}
```

Reply:
Hi jake, each connector has a `ReaderOptions` so it shouldn't related to any format. Format could be a config. You can use different deserialization schema to handle different format. You can also read the code of other connector to see how they handle different formats.
```java
public interface FileSystemReaderOptions extends ReaderOptions.BaseReaderOptions {
ConfigOption<String> FORMAT =
      key(READER_PREFIX + "format")
          .defaultValue("json");
}
```

Reply:
By looking through `FtpReaderOptions`, I can get your point. `FileSystemReaderOptions` is changed as required with 
 additional comments, plz help to review once more, thanks!